### ISSUE TITLE
global is not defined - minimal.js assumes node environment?

### ISSUE THREAD

**Issue by izhaknys** [dcodeIO/protobuf.js#874]
protobuf.js version: 6.8.0

We run this:

```shell
$ pbjs -t static-module -w es6 -o proto.js entity.proto
```

(Same thing with `-w commongjs`, by the way)

Then import it like so in a JSX:

```javascript
import { Entity } from 'proto.js'
```

Using babel, webpack emits the client code, using this babel configuration:

```json
{
    "presets":[
        "es2015", "react"
    ]
}
```

But when running in the browser, we get:

```
Uncaught ReferenceError: global is not defined
    at Object.<anonymous> (minimal.js:49)
    ...
```

Which is that line in `minimal.js`:

```javascript
util.isNode = Boolean(global.process && global.process.versions && global.process.versions.node);
```

Now that makes sense, since we are in browser environment and not node.

I just wonder if we're doing something wrong, or should `minimal.js` not assume node environment?

**Comment by dcodeIO** [comment]
The sources expect a node.js environment, while the distribution files wrap the sources through browserify and uglifyjs, where the latter defines an outer iife including the `global` var ([see](https://github.com/dcodeIO/protobuf.js/blob/master/lib/prelude.js#L3)]).

If you are building your own distribution files form the sources, respective configuration might be required. In this case, though, there should probably be one additional check testing for the existence of `global` for compatibility.

**Comment by izhaknys** [comment]
`typeof global !== "undefined"` is exactly what needed.

To overcome this issue, right now we add this to index.html:
```html
  <script>
      global = {}
  </script>
```

**Comment by izhaknys** [comment]
Right! This is a bigger scandal than Trump's RussiaGate, but it turns out our webpack had

    target: 'node',    

rather than

    target: 'web',

(a copy-paste mistake)

The latter just works out of the box and there's no need for the `global = {}` hack above.

So it works with Webpack.

Not sure whether this ticket should be closed or not. Adding `typeof global !== "undefined"` serves right an `isNode` predicate. 

But with nothing breaking, why fix? Also, not having `typeof global !== "undefined"` could hint that some outer configuration is wrong.

I'm going to leave this for you guys to decide.

Close at will.

**Comment by dcodeIO** [comment]
This was originally filed against 6.x, which is no longer supported, and newer versions guard `global` with `typeof global !== "undefined"`. Hence closing.
